docs/adr/2022-03-23_018-single-state.md

```diff
@@ -0,0 +1,68 @@
+---
+slug: 18
+title: | 
+  18. Single state in Hydra.Node.
+authors: []
+tags: [Draft]
+---
+
+## Status
+
+Draft
+
+## Context
+
+* Currently the `hydra-node` maintains two pieces of state during the life-cycle of a Hydra Head:
+  1. A `HeadState tx` provided by the `HydraHead tx m` handle interface and part of the `Hydra.Node` module. It provides the basis for the main `hydra-node` business logic in `Hydra.Node.processNextEvent` and `Hydra.HeadLogic.update`
+  [Creation](https://github.com/input-output-hk/hydra-poc/blob/a98e2907c4e425de2736782793383aad63132c14/hydra-node/src/Hydra/Node.hs#L256-L257), [Usage](https://github.com/input-output-hk/hydra-poc/blob/a98e2907c4e425de2736782793383aad63132c14/hydra-node/src/Hydra/Node.hs#L174)
+  2. `SomeOnChainHeadState` is kept in the `Hydra.Chain.Direct` to keep track of the latest known head state, including notable transaction outputs and information how to spend it (e.g. scripts and datums)
+  [Code](https://github.com/input-output-hk/hydra-poc/blob/a98e2907c4e425de2736782793383aad63132c14/hydra-node/src/Hydra/Chain/Direct.hs#L156-L162), [Usage 1](https://github.com/input-output-hk/hydra-poc/blob/a98e2907c4e425de2736782793383aad63132c14/hydra-node/src/Hydra/Chain/Direct.hs#L449), [Usage 2](https://github.com/input-output-hk/hydra-poc/blob/a98e2907c4e425de2736782793383aad63132c14/hydra-node/src/Hydra/Chain/Direct.hs#L414), [Usage 3](https://github.com/input-output-hk/hydra-poc/blob/a98e2907c4e425de2736782793383aad63132c14/hydra-node/src/Hydra/Chain/Direct.hs#L349-L352)
+  (There are other unrelated things kept in memory like the event history in the API server or a peer map in the network heartbeat component.)
+* The interface between the `Hydra.Node` and a `Hydra.Chain` component consists of 
+  - constructing certain Head protocol transactions given a description of it (`PostChainTx tx`):
+    ```hs
+    postTx :: MonadThrow m => PostChainTx tx -> m ()
```

Comment:
I don't think this first point is really about the interface between the `Hydra.Node` and the `Hydra.Chain`. This is really the client interface and how clients can interact with the on-chain state.

Reply:
Somehow I keep getting "confused" by this :thinking: ... Maybe just a naming question.